app/models/katello/content_view_module_stream_filter.rb

```diff
@@ -0,0 +1,26 @@
+module Katello
+  class ContentViewModuleStreamFilter < ContentViewFilter
+    CONTENT_TYPE = ModuleStream::CONTENT_TYPE
+    has_many :module_stream_rules, :dependent => :destroy, :foreign_key => :content_view_filter_id,
+                             :class_name => "Katello::ContentViewModuleStreamFilterRule"
+
+    validates_lengths_from_database
+
+    def generate_clauses(_repo)
+      return if module_stream_rules.blank?
+
+      module_stream_ids = module_stream_rules.map(&:module_stream_id)
+      module_streams_in(module_stream_ids) unless module_stream_ids.empty?
+    end
+
+    private
+
+    def module_stream_arel
+      ::Katello::ModuleStream.arel_table
```

Comment:
Why use arel in this file, does AR not provide the capability to do what you need?

Reply:
So this query is only  generating the where clause, instead of hard coding "(item in 111,24,5)" as string I feel arel table is a better approach